### PR TITLE
Start boss in cocoon and by a hit

### PR DESCRIPTION
**Pull request by Bernatmago**
**Changelog:**

- Boss starts in cocoon.
- Boss starts when its first hit after a timer (cocoon breaking/ effects will occur on that period)
- Scene starts in closer camera, when boss is first hit it goes back to original position and then enables lookahead after transition is finished.
(We can delay lookahead and block camera controls until boss starting effects finish idk what's the better option there)

**How to test:**
Play boss scene 👍

### CODE REVIEW

File: Gameplay/entities/enemies/BossController.cpp [hunk 1]
```diff
@@ -315,22 +323,34 @@ float Hachiko::Scripting::BossController::GetPlayerDistance()
 
 void Hachiko::Scripting::BossController::WaitingController()
 {
+    // Can be used to start boss by proximity, it is currently started via a hit
```
Comment (baransrc) [inline Gameplay/entities/enemies/BossController.cpp:326]: We can delete this now that we have the new design. If necessary we can always check the commit history.

**Review by baransrc — APPROVED** [review]
Thanks for the good work!!

File: Gameplay/entities/enemies/BossController.cpp (hunk 1 above)
Comment (Bernatmago) [reply Gameplay/entities/enemies/BossController.cpp:326]: Yep ur right we shouldnt leave it here

Comment (baransrc) [reply Gameplay/entities/enemies/BossController.cpp:326]: Let's remove it and then merge 👍

**Review by monikamnc — CHANGES_REQUESTED** [review]
Tested and working.
Paused merge until we have new boss scene, sorry >.<

**Comment by Bernatmago** [comment]
> Tested and working. Paused merge until we have new boss scene, sorry >.<

Seems fair, we can entirely remove the boss scene from this pr so we'll be fine. Ill look at the merge conflicts though

**Review by monikamnc — APPROVED** [review]
Unblock
<3

**Pull request merged**